shared/src/main/scala/amf/core/internal/plugins/syntax/SyamlSyntaxParsePlugin.scala

```diff
@@ -1,15 +1,52 @@
 package amf.core.internal.plugins.syntax
 
 import amf.core.client.common.{NormalPriority, PluginPriority}
+import amf.core.client.scala.errorhandling.AMFErrorHandler
 import amf.core.client.scala.parse.AMFSyntaxParsePlugin
 import amf.core.client.scala.parse.document.{ParsedDocument, ParserContext, SyamlParsedDocument}
+import amf.core.client.scala.validation.AMFValidationResult
 import amf.core.internal.parser.domain.JsonParserFactory
 import amf.core.internal.remote.Mimes
 import amf.core.internal.remote.Mimes._
 import amf.core.internal.unsafe.PlatformSecrets
-import org.yaml.model.{YComment, YDocument, YMap, YNode}
+import amf.core.internal.validation.CoreValidations.SyamlError
+import org.mulesoft.lexer.SourceLocation
+import org.yaml.model._
 import org.yaml.parser.YamlParser
 
+import scala.collection.mutable
+
+class SYamlAMFParserErrorHandler(eh: AMFErrorHandler) extends ParseErrorHandler with IllegalTypeHandler  {
+  override def handle[T](error: YError, defaultValue: T): T = {
+    eh.violation(SyamlError, "", error.error, part(error).location)
+    defaultValue
+  }
+
+  final def handle(node: YPart, e: SyamlException): Unit = handle(node.location, e)
+
+  override def handle(location: SourceLocation, e: SyamlException): Unit =
+    eh.violation(SyamlError, "", e.getMessage, location)
+
+  protected def part(error: YError): YPart = {
+    error.node match {
+      case d: YDocument => d
+      case n: YNode     => n
+      case s: YSuccess  => s.node
+      case f: YFail     => part(f.error)
+    }
+  }
+}
+
+class SyamlAMFErrorHandler(val eh: AMFErrorHandler) extends AMFErrorHandler with ParseErrorHandler with IllegalTypeHandler  {
```

Comment:
Extract to class. It is used in other places.

Reply:
Again, is a class already